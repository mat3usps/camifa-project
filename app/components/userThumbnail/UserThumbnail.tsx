interface IUserThumbnailProps {
  name?: string;
  email: string;
}

const UserThumbnail = ({ name, email }: IUserThumbnailProps) => {
  return (
    <div tabIndex={0} className="avatar btn btn-ghost btn-circle bg-base-100">
      <div className="w-10 rounded-full">
        <div className="flex h-full items-center justify-center">
          {name ? getNameInitials(name) : getEmailInitials()}
        </div>
      </div>
    </div>
  );

  function getNameInitials(name: string) {
    const initials = name.split(" ").map((str) => str.charAt(0));
    return initials.slice(0, 2).join("").toUpperCase();
  }

  function getEmailInitials() {
    return email.slice(0, 3);
  }
};

export default UserThumbnail;
