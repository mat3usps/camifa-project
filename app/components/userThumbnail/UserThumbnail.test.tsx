import { render, screen } from "@testing-library/react";
import type { ComponentProps } from "react";
import UserThumbnail from "./UserThumbnail";

const { getByText } = screen;

describe("<UserThumbnail />", () => {
  it("should match snapshot", () => {
    const { container } = getRenderer({ email: "john@doe.com" });
    expect(container).toMatchSnapshot();
  });

  it("renders with user name", () => {
    getRenderer({ email: "john@doe.com", name: "John Doe" });
    expect(getByText(/jd/i)).toBeInTheDocument();
  });

  it("renders with user email", () => {
    getRenderer({ email: "jane@von.com" });
    expect(getByText(/jan/i)).toBeInTheDocument();
  });
});

// Helpers
function getRenderer(props: ComponentProps<typeof UserThumbnail>) {
  return render(<UserThumbnail {...props} />);
}
