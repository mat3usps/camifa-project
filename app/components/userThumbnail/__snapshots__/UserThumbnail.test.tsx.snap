// Vitest Snapshot v1

exports[`<UserThumbnail /> > should match snapshot 1`] = `
<div>
  <div
    class="avatar btn btn-ghost btn-circle bg-base-100"
    tabindex="0"
  >
    <div
      class="w-10 rounded-full"
    >
      <div
        class="flex h-full items-center justify-center"
      >
        joh
      </div>
    </div>
  </div>
</div>
`;
